engine-sdk/src/near_runtime.rs

```diff
@@ -373,6 +375,68 @@ impl crate::promise::PromiseHandler for Runtime {
                         gas.as_u64(),
                     )
                 },
+                PromiseAction::Stake { amount, public_key } => unsafe {
+                    let amount = amount.as_u128();
+                    let pk: RawPublicKey = public_key.into();
+                    let pk_bytes = pk.as_bytes();
+                    exports::promise_batch_action_stake(
+                        id,
+                        &amount as *const u128 as _,
+                        pk_bytes.len() as _,
+                        pk_bytes.as_ptr() as _,
+                    )
+                },
+                PromiseAction::AddFullAccessKey { public_key, nonce } => unsafe {
+                    let pk: RawPublicKey = public_key.into();
+                    let pk_bytes = pk.as_bytes();
+                    exports::promise_batch_action_add_key_with_full_access(
+                        id,
+                        pk_bytes.len() as _,
+                        pk_bytes.as_ptr() as _,
+                        *nonce,
+                    )
+                },
+                PromiseAction::AddFunctionCallKey {
+                    public_key,
+                    nonce,
+                    allowance,
+                    receiver_id,
+                    function_names,
+                } => unsafe {
+                    let pk: RawPublicKey = public_key.into();
+                    let pk_bytes = pk.as_bytes();
+                    let allowance = allowance.as_u128();
+                    let receiver_id = receiver_id.as_bytes();
+                    let function_names = function_names.as_bytes();
+                    exports::promise_batch_action_add_key_with_function_call(
+                        id,
+                        pk_bytes.len() as _,
+                        pk_bytes.as_ptr() as _,
+                        *nonce,
+                        &allowance as *const u128 as _,
+                        receiver_id.len() as _,
+                        receiver_id.as_ptr() as _,
+                        function_names.len() as _,
+                        function_names.as_ptr() as _,
+                    )
+                },
+                PromiseAction::DeleteKey { public_key } => unsafe {
+                    let pk: RawPublicKey = public_key.into();
+                    let pk_bytes = pk.as_bytes();
+                    exports::promise_batch_action_delete_key(
+                        id,
+                        pk_bytes.len() as _,
+                        pk_bytes.as_ptr() as _,
+                    )
+                },
+                PromiseAction::DeleteAccount { beneficiary_id } => unsafe {
+                    let beneficiary_id = beneficiary_id.as_bytes();
+                    exports::promise_batch_action_delete_key(
+                        id,
+                        beneficiary_id.len() as _,
+                        beneficiary_id.as_ptr() as _,
+                    )
+                },
```

Comment:
I don't feel super comfortable having these methods implemented on `near-sdk`. While I agree it is the obvious place to have them, if they get exposed for some reason on the engine, an attacker can easily take control of the engine contract. 

Maybe we can have them under a feature such that the engine won't compile if that feature is on?

Reply:
Same argument applies to other variants (`PromiseAction::FunctionCall` / `PromiseAction::DeployContract`). And they are needed in some places of our codebase. So I guess we need to be extremely careful how these are used. Leaving the comment for visibility, but no action item on my end.

Reply:
I've added the feature flag you asked for in https://github.com/aurora-is-near/aurora-engine/pull/610/commits/ab3e2eaead2544a5289f0d32316bb504a4b756ae

I agree we need to be careful with other PromiseAction variants we do use. In a separate PR I think we should explicitly mark this function which create batch promises as `unsafe`. This way we will always need to think twice when using it.